tools/wt_verify.py

```diff
@@ -38,20 +38,112 @@
 import os
 import subprocess
 import sys
+import json
+import matplotlib.pyplot as plt
 
-def visualize(data, visualization_type):
+def print_output(output):
     """
-    Visualizing data
+    Print output for parsing script
     """
-    pass
+    str = "{"
+    for i, checkpoint in enumerate(output):
+        if i:
+            str += ","
+        str += "\n\t" + checkpoint + ": {"
+        for j, key in enumerate(output[checkpoint]):
+            value = output[checkpoint][key]
+            if j:
+                str += ","
+            str += "\n\t\t" + key + ": {\n\t\t\t" + json.dumps(value)
+            if "data" in value:
+                str += "\n\t\t\t\"data\": " + json.dumps(value["data"]) 
+            if "hs_modify" in value:
+                str += "\n\t\t\t\"hs_modify\": " + json.dumps(value["hs_modify"]) 
+            if "hs_update" in value:
+                str += "\n\t\t\t\"hs_update\": " + json.dumps(value["hs_update"]) 
+            str += "\n\t\t}"
+        str += "\n\t}"
+    str += "\n}"
+    print(str)
+
+
+def is_int(string):
+    """
+    Check if string can be converted to an integer
+    """
+    if isinstance(string, str):
+        if string.isdigit():
+            return int(string)
+    return string
 
 
-def parse_output(output):
+def string_to_iterable(line):
+    """
+    Helper that converts string to dictionaries and / or lists
     """
-    Parse the output file
+    dict = {}
+    for x in line.split(" | "):
+        kv_pair = x.split(": ", 1)
+        if kv_pair[1][0] == "[" and kv_pair[1][-1] == "]":
+            kv_pair[1] = kv_pair[1][1:-1].split(", ")
+            kv_pair[1] = list(map(lambda n: is_int(n), kv_pair[1]))
+        if kv_pair[0] == "addr": 
+            temp = kv_pair[1][0].split(": ")
+            dict.update({"object_id": is_int(temp[0]), "offset_range": temp[1], "size": is_int(kv_pair[1][1]), 
+                         "checksum": is_int(kv_pair[1][2])})
+        else:
+            dict[kv_pair[0]] = is_int(kv_pair[1])
+    return dict
+
+
+def parse_node(line, output, checkpoint_name, cur_node, cur_node_id):
+    line = line[len("- "):-1]
+    page_type = ([line.split(": ")[0]] + line.split(": ")[1].split())[-1]
+    if page_type == "root":
+        cur_node_id = is_int(line.split(": ")[0])
+    elif page_type == "internal" or page_type == "leaf":
+        output[checkpoint_name][cur_node_id] = cur_node 
+        cur_node_id = is_int(line.split(": ")[0])
+    else:
+        pass
+    return [cur_node_id, page_type]
+
+def parse_output(file_path):
     """
+    Parse the output file of dump_pages
+    """
+    separator = "=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=\n"
+    f = open(file_path, "r")
+    output = {}
+    line = f.readline() # separator
+    cur_node = {}
+    cur_node_id = None
+
+    while line:
+        if line == separator:
+            line = f.readline() # checkpoint
+            checkpoint_name = line.split(", ")[-1].split(": ")[-1][:-1]
```

Comment:
I find it difficult to follow the splits and slices. Could this be achieved with an regex?